Added an intro story scene

ericp275 opened the pull request:
The Story scene comes between the menu and HUB scenes. It gives a brief overview of the game. You can click through to speed up the scene. 

The dungeon loading page now displays a description of the mission where the loading text was. If you click through all the mission text, the text will go back to "loading.."

pagewin commented on fiscal-shock/Assets/Scripts/UserInterface/LoadingScreen.cs:
```diff
@@ -66,7 +93,16 @@ public class LoadingScreen : MonoBehaviour {
+        if(story){
+            async.allowSceneActivation = false;
+        }
+        if(StateManager.selectedDungeon == (DungeonTypeEnum)0){
```
```csharp
switch (StateManager.selectedDungeon) {
	case DungeonTypeEnum.Temple:
		stories = stories0;  // should be renamed to templeStory or something less vague
		break;
	case DungeonTypeEnum.Mines:
		stories = stories1;  // ditto
		break;
}
```

ericp275 commented:
> 1. Bot animation cutscene is great
> 2. I like the new hat, but should both the bank teller and the shopkeeper be wearing it?
> 3. The canvas needs to be set to scale with screen size, not constant pixel size, so it will resize automatically for different resolutions. The text size needs to stay constant, though, other the size changes as the text is typed out
> 4. The story text overflows the sides of the screen so I can't read it without resizing it
> 5. Loading screen text for the mines at least appears to get cut off. There's just one sentence
> 6. Loading screen text for the level shows again when you're going back to the hub. How about only displaying it when you first enter a dungeon and not when going back to the hub or going deeper? Player has to click through it every time right now
> 7. I thought there would be a short tutorial area where the player is taught the controls (1, 2 to access weapons, LMB to shoot, etc.), are we not doing that? If not a tutorial, then a canvas that only appears the first time you enter the hub would work (same as the dungeon tutorial canvas)
> 8. The merge conflict might be painful, what did you change in the dungeon scene? Just the loading camera?

I think I solved all of these problems. Here is what I did:

1. Thanks :)
2. I took the hat off the bank teller. (I couldn't actually take it off without editing the prefab, so instead I just made the width and height of the hat 0.)
3. Making it autosize made the text change sizes a lot, so instead I calculate and set the font size based on the screen width in the start method. This works fine but as long as all the stings of text for the story are about the same length.
4. Story text should not overflow if I got the calculations right in the start method. I tried it on a couple different screen widths and it worked fine.
5. I moved the anchor points on the text into the canvas, so it doesn't overflow anymore.
6. Should be fixed now. The story will only play when it is refreshed. Clicking the buttons on the dungeon entry canvas will refresh the story.
7. I added a tutorial to explain the controls.
8. I played around with some stuff because I didn't understand how the loading screen worked. I don't think I kept any changes to the dungeon scene. If you can set it back to how it was before this branch I think my scripts will still work fine.


I added a lot of convoluted code today, so there are probably some bugs in there. But it seems to be working so far.

pagewin commented:
All the calculations to set the font size and transform size aren't necessary. Just needed to make the canvas set to **Scale with screen size** and everything else is taken care of when you size your things right in the editor. TMP text gets resized so even "size 16" looks (relatively) the same size on 800x600 or 1920x1080
